config/_common_search_paths.py

```diff
@@ -0,0 +1,50 @@
+import os
+
+def charm_path_search(home):
+	# function that searches for charm_path in locations common across
+	# multiple architectures
+	charm_path = os.getenv('CHARM_HOME')
+
+	if (charm_path is None) and (home is not None):
+		if os.path.isdir(home + '/Charm/charm'):
```

Comment:
This function and `grackle_path_search` could be simplified a bit by creating a list of all possible paths, looping over them, and breaking when the right one is found.